mindsdb/integrations/mysql_handler/mysql_handler/mysql_handler.py

```diff
@@ -0,0 +1,172 @@
+from typing import Union, List
+import mysql.connector
+from contextlib import closing
+
+from mindsdb_sql import parse_sql
+from mindsdb.integrations.libs.base_handler import DatabaseHandler
+
+
+class MySQLHandler(DatabaseHandler):
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name)
+        self.connection = None
+        self.mysql_url = None
+        self.parser = parse_sql
+        self.dialect = 'mysql'
+        self.host = kwargs.get('host')
+        self.port = kwargs.get('port')
+        self.user = kwargs.get('user')
+        self.database = kwargs.get('database')  # todo: may want a method to change active DB
+        self.password = kwargs.get('password')
+        self.ssl = kwargs.get('ssl')
+        self.ssl_ca = kwargs.get('ssl_ca')
+        self.ssl_cert = kwargs.get('ssl_cert')
+        self.ssl_key = kwargs.get('ssl_key')
+        self.connect()
+
+    def connect(self):
+        config = {
+            "host": self.host,
+            "port": self.port,
+            "user": self.user,
+            "password": self.password
+        }
+        if self.ssl is True:
+            config['client_flags'] = [mysql.connector.constants.ClientFlag.SSL]
+            if self.ssl_ca is not None:
+                config["ssl_ca"] = self.ssl_ca
+            if self.ssl_cert is not None:
+                config["ssl_cert"] = self.ssl_cert
+            if self.ssl_key is not None:
+                config["ssl_key"] = self.ssl_key
+
+        self.connection = mysql.connector.connect(**config)
+        return self.connection
+
+    def check_status(self):
+        try:
+            con = self.connection
+            with closing(con) as con:
+                connected = con.is_connected()
+        except Exception:
+            connected = False
+        return connected
+
+    def run_native_query(self, query_str):
+        if not self.check_status():
+            self.connect()
+        try:
+            with closing(self.connection) as con:
+                cur = con.cursor(dictionary=True, buffered=True)
+                cur.execute(f"USE {self.database};")
+                cur.execute(query_str)
+                res = True
+                try:
+                    res = cur.fetchall()
```

Comment:
Should we continue using `pandas.read_sql` per each datasource and always return dataframes? Or even maybe implement that in the `Database.run_native_query`  and just accept only connection? Note important, maybe we can do that in another PR(just worth mentioning)

Reply:
I'm don't think we've settled on the output type for `run_native_query` yet.  Also inside this method, any handler could use `pd.read_sql` internally, but the input should be a raw query string.

Do you think I should change it to use `pd.read_sql` in this case?

Reply:
We can start with the current impl and then decide :raised_hands: